src/components/interactive-builder/add-question.modal.tsx

```diff
@@ -310,6 +386,12 @@ const AddQuestionModal: React.FC<AddQuestionModalProps> = ({
     setSelectedProgram(selectedItem);
     setProgramWorkflows(selectedItem?.allWorkflows);
   };
+  useEffect(() => {
+    if (isCreatingQuestion) {
+      createQuestion();
+      setIsCreatingQuestion(false);
+    }
+  }, [isCreatingQuestion, createQuestion]);
```

Comment:
The use effect is used due to react's state management when combining the added answers and the selected answers. Saving without checking when the state is changed only saves the selected answers and not the update. I tried not using it but react's state management did not allow me. The call back is used to execute also the use effect. They are both part of the same solution

Reply:
Ok I see. So instead of having a useEffect and useCallback we could change the `handleSaveMoreAnswers` to return the answers list, like so:
```
const handleSaveMoreAnswers = () => {
    const newAnswers = addedAnswers.filter(
      (newAnswer) => !selectedAnswers.some((prevAnswer) => prevAnswer.id === newAnswer.id),
    );

    const updatedAnswers = [...selectedAnswers, ...newAnswers];
    setSelectedAnswers(updatedAnswers);
    setaddedAnswers([]);
    return updatedAnswers;
  };
```
And then in the `handleCreateQuestion`, we pass in the list to the `createQuestion` function, like so:
```
 const handleCreateQuestion = () => {
    const updatedAnswers = handleSaveMoreAnswers();
    createQuestion(updatedAnswers);
    closeModal();
  };
```
And accept change the `createQuestion` function to accept a parameter and then use that param for the object:
```
const createQuestion = (conceptAnswers) => {
    try {
      const questionIndex = schema.pages[pageIndex]?.sections?.[sectionIndex]?.questions?.length ?? 0;
      const computedQuestionId = `question${questionIndex + 1}Section${sectionIndex + 1}Page-${pageIndex + 1}`;

      const newQuestion = {
        label: questionLabel,
        ......,
        ...(conceptAnswers.length && {
            answers: conceptAnswers.map((answer) => ({
              concept: answer.id,
              label: answer.text,
            })),
          }),
      };
    }
};
```
This way we could avoid the useCallback with a massive dependency list